src/components/CheckAuth/CheckAuth.js

```diff
@@ -58,23 +81,59 @@ const CheckPinWrapper = styled(Container)`
 const ACTIVE_APP_STATE = 'active';
 const BACKGROUND_APP_STATE = 'background';
 
-class CheckPin extends React.Component<Props, State> {
+class CheckAuth extends React.Component<Props, State> {
   static defaultProps = {
     revealMnemonic: false,
   };
   state = {
     biometricsShown: false,
     lastAppState: AppState.currentState,
+    showPin: false,
   };
 
   componentDidMount() {
     addAppStateChangeListener(this.handleAppStateChange);
-    const { useBiometrics, revealMnemonic } = this.props;
+    const {
+      useBiometrics, revealMnemonic, enforcePin, modalProps,
+    } = this.props;
     const { lastAppState } = this.state;
-    if (useBiometrics
-      && !revealMnemonic
-      && lastAppState !== BACKGROUND_APP_STATE) {
-      this.showBiometricLogin();
+    // do nothing if auth isn't supposed to be checked
+    if (!(modalProps && !modalProps.isVisible)) {
+      if (useBiometrics
+        && !revealMnemonic
+        && lastAppState !== BACKGROUND_APP_STATE) {
+        this.showBiometricLogin();
+      } else if (lastAppState !== BACKGROUND_APP_STATE && !enforcePin) {
+        this.checkPrivateKey();
+      }
+    }
```

Comment:
Add lodash `get` and then:
```suggestion
    if (get(modalProps, 'isVisible')) return;
    if (useBiometrics
      && !revealMnemonic
      && lastAppState !== BACKGROUND_APP_STATE) {
      this.showBiometricLogin();
    } else if (lastAppState !== BACKGROUND_APP_STATE && !enforcePin) {
      this.checkPrivateKey();
    }
```